### ISSUE TITLE
Inputting TextInput answers in landscape auto-closes keyboard

### ISSUE THREAD

**Issue by BenHenning** [oppia/oppia-android#1769]
This is the same bug as #1767 except after its cause PR is reverted, this still repros but only for the TextInput interaction which indicates it may have a different root cause.

**Comment by BenHenning** [comment]
Note that some tests in #1630 have been disabled until this issue is resolved.

**Comment by BenHenning** [comment]
Interestingly, this sometimes comes across as initially working (e.g. keyboard does not automatically dismiss), but dismissing the keyboard manually then re-selecting the input box can trigger automatic dismissal.

**Comment by BenHenning** [comment]
This seems to not be a regression--I'm able to repro it as far back as 6722988d, and gave up looking it farther. Since we know this is specific to my device, making the choice to push this to beta. It's a bad bug for users to hit, but if they do they can work around it by rotating their device to portrait.

**Comment by anandwana001** [comment]
Keyboard closing in the landscape  in espresoo will get solved using, 
```kotlin
onView(withId(interactionViewId))
  .perform(typeText(text))
onView(isRoot()).perform(closeSoftKeyboard())
testCoroutineDispatchers.runCurrent()
```
but not in robolectric. I think we might need to create custom inputs here like - 
https://github.com/oppia/oppia-android/blob/62beb85fad2d32a33ee1dc4fc63ec074ad912a2f/app/src/sharedTest/java/org/oppia/android/app/testing/InputInteractionViewTestActivityTest.kt#L961



the major issue to solve here is dragging doesn't work in Robolectric, might be timing issue, will look into deep, just a thought as of now. 
https://github.com/oppia/oppia-android/blob/62beb85fad2d32a33ee1dc4fc63ec074ad912a2f/app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt#L1071

**Comment by BenHenning** [comment]
Note that this issue is actually happening in production builds so far as I can remember.

**Comment by yurilev** [comment]
Hi, can I work on this?

**Comment by anandwana001** [comment]
Hi, @yurilev any update on this?

**Comment by BenHenning** [comment]
@yurilev are you sure these are related? From my experience it seems like the auto-closing happens on real Android devices (& potentially not on emulators--I forget). #1612 is describing an issue that only occurs in Robolectric due to missing functionality in the Robolectric test framework.

**Comment by yurilev** [comment]
One sec, I'm writing the full reply with the context.

**Comment by yurilev** [comment]
Hi @anandwana001 and @BenHenning,

I've tried reproducing again (looked at #1767 as well), unsuccessfully. I tested on:
* About 4 mobile emulators, all running API 29. In case screen size could be related, with screens ranging from 2.7" 240x320 to 6.3" Pixel 4 XL. And one tablet.
* Physical Pixel 2 running API 29 and Nexus 5 running API 23.

Per [this comment](https://github.com/oppia/oppia-android/issues/1769#issuecomment-685999168), dismissed the keyboard in a number of different ways.

None of these reproduced the issue.

Moreover, the two skipped tests in `StateFregmentTest` are now passing in Espresso with no issue. [PR #2165](https://github.com/oppia/oppia-android/pull/1630) probably fixed them, on the same day @BenHenning last commented he reproduced it.

I'll send a PR tomorrow to re-enable these tests in Espresso. #1612 covers Robolectic (there are preexisting TODOs for it there), I'll try looking into that next.

@BenHenning - unless you can still reproduce it on your device or have other suggestions for testing, I think this should be closed.

**Comment by BenHenning** [comment]
Nice findings @yurilev. Confirming that I also can't repro the issue anymore. Bit surprised to hear that the fix correlates to #1630 since I recall reproing this after that PR, but maybe something else later on fixed the problem in conjunction with that PR.

**Comment by yurilev** [comment]
The bit about #1630 was conjecture on my part – the changes made seemed like they could fix it, and the timeline fits. I can try and test with the version prior if you want me to confirm.
